src/effects/native/tremoloeffect.cpp

```diff
@@ -0,0 +1,224 @@
+#include "effects/native/tremoloeffect.h"
+
+namespace {
+//  Used to avoid gain discontinuities when changing parameters too fast
+constexpr double kMaxGainIncrement = 0.001;
+
+constexpr int kNumberOfChannels  = 2;
+}
+
+// static
+QString TremoloEffect::getId() {
+    return "org.mixxx.effects.tremolo";
+}
+
+// static
+EffectManifest TremoloEffect::getManifest() {
+    EffectManifest manifest;
+    manifest.setId(getId());
+    manifest.setName(QObject::tr("Tremolo"));
+    manifest.setShortName(QObject::tr("Tremolo"));
+    manifest.setAuthor("The Mixxx Team");
+    manifest.setVersion("1.0");
+    manifest.setDescription("An amplitude modulation effect");
+
+    EffectManifestParameter* rate = manifest.addParameter();
+    rate->setId("rate");
+    rate->setName(QObject::tr("Rate"));
+    rate->setShortName(QObject::tr("Rate"));
+    rate->setDescription(QObject::tr("Controls the rate of the effect\n"
+    "every 4 beats - 1/8 beat if tempo is detected (decks and samplers) \n"
+    "1/4 Hz - 8 Hz if no tempo is detected (mic & aux inputs, master mix)"));
+    rate->setControlHint(
+        EffectManifestParameter::ControlHint::KNOB_LOGARITHMIC);
+    rate->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    rate->setUnitsHint(EffectManifestParameter::UnitsHint::BEATS);
+    rate->setDefaultLinkType(EffectManifestParameter::LinkType::LINKED);
+    rate->setDefault(1);
+    rate->setMinimum(1.0/4);
+    rate->setMaximum(8);
+
+    EffectManifestParameter* shape = manifest.addParameter();
+    shape->setId("shape");
+    shape->setName(QObject::tr("Shape"));
+    shape->setShortName(QObject::tr("Shape"));
+    shape->setDescription(QObject::tr("Sets the duty cycle of the modulation"
+    "10% - 90% of the effect period"));
+    shape->setControlHint(EffectManifestParameter::ControlHint::KNOB_LINEAR);
+    shape->setSemanticHint(EffectManifestParameter::SemanticHint::UNKNOWN);
+    shape->setUnitsHint(EffectManifestParameter::UnitsHint::UNKNOWN);
+    shape->setMinimum(0.1);
+    shape->setDefault(0.5);
+    shape->setMaximum(0.9);
+
+    EffectManifestParameter* smooth = manifest.addParameter();
+    smooth->setId("smooth");
```

Comment:
I think this is normally called "wave".
If you have an other hardware model with other names it works also me. I just want to make sure we do not create our own namespace.

Reply:
Most stompboxes come with a small drawing for what I called "shape" & "smooth" parameter
We can use what the Shapeshifter from seymour duncan uses : [Shapeshifter](https://www.google.fr/search?q=shapeshifter+seymour+duncan&client=ubuntu&channel=fs&dcr=0&source=lnms&tbm=isch&sa=X&ved=0ahUKEwj_gczdgPvXAhXCIlAKHZ6xBdkQ_AUICygC&biw=1533&bih=769#imgrc=DQUxHyxZeMJpgM:)
- "Smooth" would be called Wave (or Waveform, more explicit without any drawing)
- "Shape" would stay the same

Reply:
OK